client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -37,11 +37,14 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 	const history = useHistory();
 	const { search } = useLocation();
 	const { setStepData } = useDispatch( STEPPER_INTERNAL_STORE );
+	const intent = useSelect( ( select ) => select( ONBOARD_STORE ).getIntent() );
 	const stepNavigation = flow.useStepNavigation( currentRoute, async ( path, extraData = null ) => {
 		// If any extra data is passed to the navigate() function, store it to the stepper-internal store.
-		if ( extraData ) {
-			setStepData( extraData );
-		}
+		setStepData( {
```

Comment:
Also, I can't work out where we use extraData in our code at the moment?

Reply:
Yes, `extraData` and Internal stepper store are defined but never used before, that's why I think it is good to store `intent` value there

Reply:
> I think you also need to change the retrieval to use the new data format `{ extraData: extraData }` or potentially use spread syntax `...extraData` to combine the objects at

`extraData` is [potential null](https://github.com/Automattic/wp-calypso/blob/c04d343e7c51ec3bc0f180892ab388c554601996/packages/data-stores/src/stepper-internal/actions.ts#L1), that why I assign it to a property, should we update params type for `setStepData`?

Reply:
I think what @roo2 is suggesting is to use spread operator: 

```
setStepData( { path, intent, ...extraData } );
```

Otherwise, you'd be storing the object `extraData` under the key `extraData`.
Say, for example, `extraData = { a: 1 }`, it would result in: ` { intent: 'write', extraData: { a: 1 } }`.

By using the spread operator, it would result it ` { intent: 'write', a: 1 }`